src/org/intellij/plugins/ceylon/ide/startup/AbstractMetamodelEnricher.java

```diff
@@ -29,7 +29,8 @@
 abstract class AbstractMetamodelEnricher implements ApplicationComponent {
 
     private static Map<String, String> registeredModules = new HashMap<>();
-    private static final Pattern moduleArchivePattern = Pattern.compile("(.+)-([^\\-]+)\\.(j|J|c|C)ar");
+    private static final Pattern moduleArchivePatternCar = Pattern.compile("([^\\-]+)-(.+)\\.(c|C)ar");
+    private static final Pattern moduleArchivePatternJar = Pattern.compile("(.+)-([^\\-]+)\\.(j|J)ar");
```

Comment:
Let's hope we won't have to use something like `org.antlr.antlr4-runtime-osgi-4.5.1-2.jar` in the future :)